src/components/List.jsx

```diff
@@ -18,10 +19,30 @@ export default function List() {
 
   const handleCheckBox = (e, item) => {
     e.preventDefault();
-    let now = Timestamp.now().seconds;
+
     const docItem = doc(db, userToken, item.id);
+    let now = Timestamp.now().seconds;
+    const lastPurchase = item.data().lastPurchaseDate;
+    const itemCreated = item.data().dateItemAdded;
+    const totalPurchases = item.data().totalPurchases + 1;
+
+    let daysSinceLastTransaction;
+    if (lastPurchase === null) {
+      daysSinceLastTransaction = Math.round((now - itemCreated) / 86400);
+    } else {
+      daysSinceLastTransaction = Math.round((now - lastPurchase) / 86400);
+    }
```

Comment:
This might be "too complex", but this is a possible use case for the [`??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

ie. `daysSinceLastTransaction = Math.round((now - (lastPurchase ?? itemCreated)) / 86400);`

Reply:
I didn't know about the ?? operator- thank you, @adidalal !! Just to check my understanding...
If the expression on the left side of the ?? operator evaluates to null or undefined, return the value on the right side. Otherwise return the value on the left. 
As long as I know what ?? does, I don't think it seem too complex. It's nice to turn 5 lines of code into 1! 😸

Reply:
yep, that's exactly correct

Reply:
Thank you for that explanation and suggestion! That makes sense and is very helpful.